### PR TITLE
Split out our directory structure vs. Forge

### PR DESCRIPTION
**Pull request by Zidane**
After discussion with leadership, we have elected to split out the directory structure of Sponge to remain compatible (mostly) with 3rd party platforms (namely Forge). Current running idea is to do as follows:

* `../mods` -> `../plugins`
* Split where plugin configs and Sponge configs are stored
* Remove the alternate plugins directory configuration property. The only reason this was useful was SpongeForge co-mingling plugins and mods. In the high unlikelihood that an implementation is made that the root platform also has `plugins`, we can circle back at that point.
* Remove specifying where the config directory for plugins are located. Per the same reason as the prior bullet.
* Add `../plugin-configs`
* Add `../sponge`. This will become where Sponge's internal files will mostly reside (except for things specific to worlds)
  * Add `../sponge/config`
  * Add `../sponge/libraries`

### CODE REVIEW

**Comment by ryantheleach** [comment]
I'm curious as to what this change means for what was previously called "hybrid" Sponge plugins. e.g. they contained both mods and plugins , usually in the form of mixins.

File: src/applaunch/java/org/spongepowered/common/applaunch/CorePaths.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Core directory structure of the Sponge platform
+ *
+ * <base_directory>
+ *  - plugins
+ *  - plugin-configs
```
Comment (Yeregorix) [inline src/applaunch/java/org/spongepowered/common/applaunch/CorePaths.java:36]: IMO `plugins/configs/` would be more consistent with `sponge/configs/`.

Comment (Zidane) [reply src/applaunch/java/org/spongepowered/common/applaunch/CorePaths.java:36]: My goal was to limit drill downs and keep the most commonly accessed folders quick from the root.

**Comment by Zidane** [comment]
> I'm curious as to what this change means for what was previously called "hybrid" Sponge plugins. e.g. they contained both mods and plugins , usually in the form of mixins.

Honestly it would be best to deploy that in our plugins directory. Mixins are now read from plugin jars in the plugins directory in 1.16.5+

**Comment by ImMorpheus** [comment]
Hey! You seem to be running an unsupported version (API 8) of SpongeAPI. Support for API 8 ended in October 2023.

API-8 is now end-of-life. We will not be issuing any more API updates for version 8 of our API[^1]

As part of our legacy cleanup we are closing issues relating to API8/1.16
___

If you wish to move this forward, please rebase this PR on the current branch and reopen it.

[^1]: https://forums.spongepowered.org/t/sponge-status-update-17th-october-2023/42212#spongeapi-82-1

**Pull request closed without merging**